### DISCUSSION TITLE
Mix Fan curve only looking at fan speed instead of temp?

### DISCUSSION THREAD

**Discussion by IPHANT0MI** [Rem0o/FanControl.Releases#1139] [Q&A] [answered]
Hello Remi, 

First of all AMAZING work on this project! Please never stop tinkering and innovating! 

Issue: Maybe it's just me but when I setup my  a "Mix" fan curve and setup it to a "Max" function between CPU/GPU i noticed that the mix fan curve only looks at the fan RPM of each one (CPU/GPI) and not the actually temp.

For example, if i like my GPU fan to always been spinning at 50% of the fan speed but my temp of my GPU is lower than the CPU, it will favor the GPU temp and not the CPU.

Not sure if i am doing anything wrong but i feel the mix curve is not looking at temp speeds but fans speeds to determine the which one it should be looking at.

Thoughts? Yes, i did see a [post](https://github.com/Rem0o/FanControl.Releases/discussions/681) However, based on your answer are you saying there is no solution to this and Max only looks at fan speed and not temp? If so, is there a potential solution in the future? 

![Screenshot 2022-09-27 215741](https://user-images.githubusercontent.com/91971743/192684925-721ac860-4738-4c99-87a2-4f93f29f1b56.jpeg)

**Comment by Rem0o** [comment, marked as answer]
Well yeah, the mix curve take the max output of the selected curves.  If you do want it to act as a max of temperature, you can adjust your curves so that higher temp always equals higher %.  

Looking at temp wouldn't really make sense. First because the mix fan curve doesn't "see" the underlying temperature of its selected fan curves, it only sees the output. In your scenario, if it looked at temperature, it would output the lowest % of the 2, which is super counter-intuitive.  Also, the temperature range for 2 sources can be very different. A hard drive can hover between 25C to 40C, while a CPU can hover between 45C to like 90C. If you mixed those two, the CPU curve would always win, even though the hard drive might be cooking itself. The absolute comparison of 2 different temperatures in that regards doesn't make sense. The % of a curve indicates how "hot" a specific temperature is, if you set those correctly.  In my example, I would set a 100% for 40C on the drive, and 100% at 90C on the CPU.

If you do want a max of temperature, you can create a mix sensor, which will output the maximum temperature of multiple sources, which you can then input into a curve.

**Reply by IPHANT0MI** [reply]
That makes sense! Thank you for the reply!
